data/1.0.0-preview-1/schema/ciphertext_ballot.schema.json

```diff
@@ -1,4 +1,392 @@
+          "oneOf": [
+            { "type": "null" },
+            { "$ref": "#/definitions/DisjunctiveChaumPedersenProof" }
+          ],
+          "default": null
+        }
+      }
+    },
+    "CiphertextBallotContest": {
+      "title": "Ciphertext Ballot Contest",
+      "description": "A CiphertextBallotContest represents the selections made by a voter for a specific ContestDescription.",
+      "type": "object",
+      "required": [
+        "object_id",
+        "sequence_order",
+        "description_hash",
+        "ballot_selections",
+        "ciphertext_accumulation",
+        "crypto_hash",
+        "nonce",
+        "proof",
+        "extended_data"
+      ],
+      "properties": {
+        "object_id": {
+          "title": "Object Id",
+          "type": "string"
+        },
+        "sequence_order": {
+          "title": "Sequence Order",
+          "description": "Used for ordering in a ballot to ensure various encryption primitives are deterministic.",
+          "$ref": "#/definitions/SequenceOrder"
+        },
+        "description_hash": {
+          "title": "Description Hash",
+          "description": "Hash from contestDescription",
+          "$ref": "#/definitions/ElementModQ"
+        },
+        "ballot_selections": {
+          "title": "Ballot_selections",
+          "type": "array",
+          "default": [],
+          "items": {
+            "title": "Items",
+            "$ref": "#/definitions/CiphertextBallotSelection"
+          }
+        },
+        "ciphertext_accumulation": {
+          "title": "Ciphertext Accumulation",
+          "description": "The encrypted representation of all of the vote fields (the contest total)",
+          "$ref": "#/definitions/ElGamalCiphertext"
+        },
+        "crypto_hash": {
+          "title": "Crypto Hash",
+          "description": "Hash of the encrypted values",
+          "$ref": "#/definitions/ElementModQ"
+        },
+        "nonce": {
+          "title": "Nonce",
+          "description": "The nonce used to generate the encryption. Sensitive & should be treated as a secret",
+          "oneOf": [
+            { "type": "null" },
+            { "$ref": "#/definitions/ElementModQ" }
+          ],
+          "default": null
+        },
+        "proof": {
+          "title": "Proof",
+          "description": "The proof demonstrates the sum of the selections does not exceed the maximum available selections for the contest, and that the proof was generated with the nonce",
+          "oneOf": [
+            { "type": "null" },
+            { "$ref": "#/definitions/ElGamalCiphertext" }
+          ],
+          "default": null
+        },
+        "extended_data": {
+          "title": "Extended Data",
+          "description": "Encrypted representation of the extended_data field",
+          "oneOf": [
+            { "type": "null" },
+            { "$ref": "#/definitions/HashedElGamalCiphertext" }
```

Comment:
This produces an error in the JSON Schema Validator I was using because `HashedElGamalCiphertext` is nested under `ElGamalCiphertext` not under `definitions`:

```
Error parsing schema
Message:
Error when resolving schema reference '#/definitions/HashedElGamalCiphertext'. Path 'definitions.CiphertextBallotContest.properties.extended_data.oneOf[1]', line 333, position 13.
```